rust/otap-dataflow/crates/otap/src/attribute_processor.rs

```diff
@@ -0,0 +1,643 @@
+                // Other actions not yet supported
+                _ => {}
+            }
+        }
+
+        Self {
+            transform: AttributesTransform {
+                rename: if renames.is_empty() {
+                    None
+                } else {
+                    Some(renames)
+                },
+                delete: if deletes.is_empty() {
+                    None
+                } else {
+                    Some(deletes)
+                },
+            },
+        }
+    }
+}
+
+#[async_trait(?Send)]
+impl local::Processor<OtapPdata> for AttributeProcessor {
+    async fn process(
+        &mut self,
+        msg: Message<OtapPdata>,
+        effect_handler: &mut local::EffectHandler<OtapPdata>,
+    ) -> Result<(), EngineError<OtapPdata>> {
+        match msg {
+            Message::Control(_) => Ok(()),
+            Message::PData(pdata) => {
+                let signal = pdata.signal_type();
+                let mut pdata = pdata;
+
+                match pdata {
+                    OtapPdata::OtapArrowRecords(ref mut records) => {
+                        apply_transform(records, signal, &self.transform)?;
+                        effect_handler.send_message(pdata).await
+                    }
+                    OtapPdata::OtapArrowBytes(_) => {
+                        // Convert to records, apply, convert back to bytes to preserve variant
+                        let signal_ty = pdata.signal_type();
+                        let records: OtapArrowRecords = pdata.clone().try_into()?;
+                        let mut records_mut = records;
+                        apply_transform(&mut records_mut, signal_ty, &self.transform)?;
+                        let bytes: crate::grpc::OtapArrowBytes = records_mut.try_into()?;
+                        let pdata_back: OtapPdata = bytes.into();
+                        effect_handler.send_message(pdata_back).await
+                    }
+                    OtapPdata::OtlpBytes(otlp_bytes) => {
+                        // Convert to OTAP, apply transform, convert back
+                        let records: OtapArrowRecords = otlp_bytes.try_into()?;
+                        let mut records_mut = records;
+                        apply_transform(&mut records_mut, signal, &self.transform)?;
+                        let bytes: crate::pdata::OtlpProtoBytes = records_mut.try_into()?;
+                        let pdata_back = OtapPdata::OtlpBytes(bytes);
+                        effect_handler.send_message(pdata_back).await
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[allow(clippy::result_large_err)]
+fn apply_transform(
+    records: &mut OtapArrowRecords,
+    signal: SignalType,
+    transform: &AttributesTransform,
+) -> Result<(), EngineError<OtapPdata>> {
+    let payloads = attrs_payloads_for_signal(signal);
+
+    // Only apply if we have transforms to apply
+    if transform.rename.is_some() || transform.delete.is_some() {
```

Comment:
This is not a big deal, but in the unusual event that this were true (e.g. user configures this processor, but doesn't configure any actions), it would actually be more efficient to do this check before deserializing the data to OTAP. Kind of like what I was suggesting in this comment:
https://github.com/open-telemetry/otel-arrow/pull/941/#discussion_r2277628516